src/hooks/useArNSIntentPrice.tsx

```diff
@@ -0,0 +1,51 @@
+import { TurboArNSIntentPriceParams } from '@src/services/turbo/TurboArNSClient';
+import { useWalletState } from '@src/state';
+import { useQuery } from '@tanstack/react-query';
+
+import { useTurboArNSClient } from './useTurboArNSClient';
+
+export function useArNSIntentPrice({
+  address,
+  name,
+  intent,
+  increaseQty,
+  type,
+  years,
+  promoCode,
+}: Omit<TurboArNSIntentPriceParams, 'address' | 'currency'> & {
+  address?: string;
+}) {
+  const turbo = useTurboArNSClient();
+  const [{ walletAddress }] = useWalletState();
+
+  const userAddress = address ?? walletAddress;
+
+  return useQuery({
+    queryKey: [
+      'turbo-arns-intent-price',
+      userAddress?.toString(),
+      name,
+      intent,
+      increaseQty,
+      type,
+      years,
+      promoCode,
+    ],
+    queryFn: async () => {
+      if (!turbo || !userAddress) {
+        throw new Error('Turbo client or address is not available');
+      }
+      return turbo.getPriceForArNSIntent({
+        address: userAddress.toString(),
+        name,
+        intent,
+        increaseQty,
+        type,
+        years,
+        promoCode,
+      });
+    },
+    enabled: !!turbo && !!userAddress,
+    staleTime: 1000 * 30,
```

Comment:
The UI has 5 minutes for refresh time but here it's 30 seconds; should match. 
<img width="470" alt="image" src="https://github.com/user-attachments/assets/d3f2aee4-383d-4388-ab3f-f648b36a5d00" />

Reply:
or comment if I'm misunderstanding something here

Reply:
[54c3c83](https://github.com/ar-io/arns-react/pull/734/commits/54c3c83ea72c32780833177831d1405cae1b310e)